### PR TITLE
RFC possible implementation for user override of transpile options 

### PR DESCRIPTION
**Pull request by PopGoesTheWza**
@grant please advise

### CODE REVIEW

**Comment by PopGoesTheWza** [comment]
Ah... I used settings unsuitable for node below 8.10.0 :(

File: package.json [hunk 1]
```diff
@@ -14,15 +15,23 @@
...
   "license": "MIT",
+  "engines": {
+    "node": ">=8.10.0"
+  },
   "bugs": {
     "url": "https://github.com/grant/ts2gas/issues"
   },
   "homepage": "https://github.com/grant/ts2gas#readme",
   "dependencies": {
+    "lodash": "^4.17.11",
```
Comment (grant) [inline package.json]: Use es6 features instead of lodash.

File: package.json [hunk 2]
```diff
@@ -14,15 +15,23 @@
...
     "typescript": "^3.2.2"
   },
   "devDependencies": {
     "tslint": "^5.11.0",
+    "@types/lodash": "^4.14.119",
     "@types/node": "^10.12.18"
+  },
+  "peerDependencies": {
+    "@google/clasp": "^1.7.0"
```
Comment (grant) [inline package.json]: We don't need this peer dep.

File: src/index.ts [hunk 3]
```diff
@@ -113,26 +114,57 @@ const ts2gas = (source: string) => {
...
+      // pretty: true,
+    },
+  };
+
+  /** These the settings are always used */
+  const statics: ts.TranspileOptions = {
+    compilerOptions: {
+      target: ts.ScriptTarget.ES3,
+      noLib: true,
+      // lib: [  // Not be relevant with ts.transpileModule(). also 'noLib' is true
```
Comment (grant) [inline src/index.ts]: We don't need this comment.

File: tsconfig.json [hunk 4]
```diff
@@ -1,17 +1,12 @@
...
-    "target": "es5",
-    "lib": [
-      "es5",
-      "es6",
-      "es2015",
-      "es2016",
-      "es2017"
-    ],
+    "declaration": true,
+    "target": "es2017",  // fully supported with node ^8.10.0 (cf. https://node.green/)
```
Comment (grant) [inline tsconfig.json]: I think only es5 will work with Apps Script.

File: package.json (hunk 1 above)
Comment (PopGoesTheWza) [reply package.json]: OK. I just didn't want to re-invent the wheel and re-implement deep-assign (merge/mergeWith)

File: package.json (hunk 2 above)
Comment (PopGoesTheWza) [reply package.json]: TBH, I wasn't sure if it would help at all :)

File: src/index.ts [hunk 5]
```diff
@@ -113,26 +114,57 @@ const ts2gas = (source: string) => {
...
-  });
+  };
+
+  // keep only overridable properties
+  transpileOptions = _.isObject(transpileOptions)
+    ? _.pick(transpileOptions, ['compilerOptions', 'renamedDependencies'])
+    : {};
+
+  // merge properties in order for proper override
+  transpileOptions = _.merge({},
```
Comment (grant) [inline src/index.ts]: https://stackoverflow.com/a/25877818

File: tsconfig.json (hunk 4 above)
Comment (PopGoesTheWza) [reply tsconfig.json]: this is the config used on ts2gas.ts (and test.ts), hence the target is node (^8.10.0) not GAS

Comment (grant) [reply tsconfig.json]: Oh, thats fine.

File: src/index.ts (hunk 5 above)
Comment (PopGoesTheWza) [reply src/index.ts]: Object.assign only work on the direct keys and does not recurse on values which are objects.

File: src/index.ts (hunk 3 above)
Comment (PopGoesTheWza) [reply src/index.ts]: Gone (but I added one about renamedDependencies for documentation/reference)

File: src/index.ts (hunk 5 above)
Comment (PopGoesTheWza) [reply src/index.ts]: I have no problem replacing _.pick;) with plain ES. Same for a deepAssign() which recurse inner objects and concatenate inner arrays.
Please just confirm this.

**Comment by PopGoesTheWza** [comment]
FYI, clasp already has (indirect dependencies with lodash) 
image
:)

**Comment by grant** [comment]
Thanks.
Trying to keep direct dependencies to a minimum.

**Pull request merged**